docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,369 @@
+
+The primary metric used for scaling should be the number of shoots already scheduled onto that seed either as a direct value or as a percentage of the seed's capacity for shoots introduced in [Ensuring Seeds Capacity for Shoots Is Not Exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded) (*utilization*). Later, custom metrics based on other resources, including provider-specific resources, could be considered as well.
+
+**Note:** Even if the controller is called *Horizontal Pod Autoscaler*, it is capable of scaling any resource with a `scale` subresource, using any custom metric. Therefore, initially it was proposed to use this controller directly. However, a number of important drawbacks were identified with this approach, and so it is no longer proposed here.
+
+### SeedAutoscaler
+
+The SeedAutoscaler automatically scales the number of [ShootedSeeds](#shootedseeds) in a [ShootedSeedSet](#shootedseedsets) based on observed resource utilization. The resource could be any resource that is tracked via the `capacity` and `allocatable` fields in the Seed status, including in particular the number of shoots already scheduled onto the seed.
+
+#### SeedAutoscaler Resource
+
+The SeedAutoscaler is implemented as a custom resource and a new controller. The resource determines the behavior of the controller. The `SeedAutoscaler` resource has a `scaleTargetRef` that specifies the target resource to be scaled, the minimum and maximum number of replicas, as well as a list of metrics. The only supported metric type initially is `Resource` for resources that are tracked via the `capacity` and `allocatable` fields in the Seed status. The resource target can be of type `Utilization` or `AverageValue`.
+
+Example `SeedAutoscaler` resource:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: SeedAutoscaler
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  scaleTargetRef:
+    apiVersion: seedmanagement.gardener.cloud/v1alpha1
+    kind: ShootedSeedSet
+    name: crazy-botany
+  minReplicas: 1
+  maxReplicas: 10
+  metrics:
+  - type: Resource # Only Resource is supported
+    resource:
+      name: shoots
+      target:
+        type: Utilization # Utilization or AverageValue
+        averageUtilization: 50
+```
+
+#### SeedAutoscaler Controller
+
+`SeedAutoscaler` resources are reconciled by a new controller, either in `gardenlet` or out-of-tree, similarly to [cluster-autoscaler](https://github.com/gardener/autoscaler). The controller periodically adjusts the number of replicas in a ShootedSeedSet to match the observed average resource utilization to the target specified by user.
```

Comment:
I guess, there is no reason to do this out-of-tree, this can directly be done in the `gardener-controller-manager`.
The `gardenlet` wouldn't be the right component FMPOV, as the to-be-scaled objects (i.e. `ShootedSeedSets`) are also controlled centrally per gardener installation by the `gardener-controller-manager`.

Reply:
Yes, you are right, `gardenlet` is definitely wrong here, just missed to change it to GCM when I last revised the document. I mention out-of-tree because I got the impression @rfranzke seems to be in favor of keeping stuff in independent controllers where it makes sense, rather than overloading GCM and gardenlet with more and more responsibilities.

Reply:
Changed to `gardenlet` to `gardener-controller-manager`, but otherwise kept "or out-of-tree" due to the above considerations. I guess we should have another discussion when we come to the point of implementing it.